processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -721,30 +721,33 @@ else if ( mapping.getJavaExpression() != null ) {
                     sourceRef = getSourceRef( method.getSourceParameters().get( 0 ), targetPropertyName );
                 }
 
-                if ( sourceRef.isValid() ) {
+                if ( sourceRef != null ) {
+                    // sourceRef == null is not considered an error here
```

Comment:
Maybe add an example when this can happen. For example for `@Mapping(target = "name", defaultValue = "test")`. Perhaps update the test since I was confused initially :).

Reply:
target without source can happen in many ways :smile: . It's a hook. In this case to apply a qualifier on a name based mapping. An example is a good idea.. but not as such scope of this PR.. Its already there since ages..

I'll add a separate PR for that..

Need to check why it's not compiling..